internal/wire/relationships.go

```diff
@@ -87,3 +92,57 @@ func RemoveFromGraph(context interface{}, store *cayley.Handle, quadParams []Qua
 	log.Dev(context, "RemoveFromGraph", "Completed")
 	return nil
 }
+
+// InferRelationships infers realtionships based on patterns corresponding to
+// types of items.
+func InferRelationships(context interface{}, mgoDB *db.DB, items []bson.M) ([]QuadParams, error) {
+	log.Dev(context, "InferRelationships", "Started : %d Items", len(items))
+
+	// Infer relationships for each provided item.
+	var quadParams []QuadParams
+	for _, item := range items {
+
+		// Get the relevant pattern.
+		itemType := item["type"]
+		p, err := pattern.GetByType(context, mgoDB, itemType.(string))
+		if err != nil {
+			continue
+		}
+
+		// Loop over inferences in the pattern.
+		data := item["data"]
+		dataMap, ok := data.(map[string]interface{})
```

Comment:
Let's talk about this function, I want to understand why we are not using a type.